libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt

```diff
@@ -96,65 +137,188 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
+     * @param legIndex new leg index
+     *
+     * @return an initialized route state as [NavigationStatus]
+     */
     override fun updateLegIndex(routeIndex: Int, legIndex: Int): NavigationStatus =
         navigator.changeRouteLeg(routeIndex, legIndex)
 
     // Free Drive
-
+    /**
+     * Uses routing engine and local tile data to generate electronic horizon json.
+     *
+     * Consumes a list of points, matches them to the routing graph
+     * (i.e. does traceAttributes) and prolongs this path
+     * in selected directions (one way, one way with branches, all ways)
+     * according to the provided eHorizon distance (the speed is derived from input points).
+     *
+     * @param request the uri used when hitting the http service
+     *
+     * @return a [RouterResult] object with the json and a success/fail bool
+     */
     override fun getElectronicHorizon(request: String): RouterResult =
         navigator.getElectronicHorizon(request)
 
     // Offline
 
+    /**
+     * Caches tiles around last set route
+     */
     override fun cacheLastRoute() {
         navigator.cacheLastRoute()
     }
 
+    /**
+     * Configures routers for getting routes offline.
+     *
+     * @param routerParams Optional [RouterParams] object which contains router configurations for
+     * getting routes offline.
+     * @param httpClient A platform specific [HttpInterface]. Can be null so default
+     * implementation will be used.
+     *
+     * @return number of tiles founded in the directory
+     */
     override fun configureRouter(routerParams: RouterParams, httpClient: HttpInterface?): Long =
         navigator.configureRouter(routerParams, httpClient)
 
+    /**
+     * Uses valhalla and local tile data to generate mapbox-directions-api-like json.
+     *
+     * @param url the directions-based uri used when hitting the http service
+     * @return a [RouterResult] object with the json and a success/fail bool
+     */
     override fun getRoute(url: String): RouterResult = navigator.getRoute(url)
 
+    /**
+     * Passes in an input path to the tar file and output path.
+     *
+     * @param tarPath The path to the packed tiles.
+     * @param destinationPath The path to the unpacked files.
+     *
+     * @return the number of unpacked tiles
+     */
     override fun unpackTiles(tarPath: String, destinationPath: String): Long =
         navigator.unpackTiles(tarPath, destinationPath)
 
+    /**
+     * Removes tiles wholly within the supplied bounding box. If the tile is not
+     * contained completely within the bounding box it will remain in the cache.
+     * After removing files from the cache any routers should be reconfigured
+     * to synchronize their in memory cache with the disk.
+     *
+     * @param tilePath The path to the tiles.
+     * @param southwest The lower left coord of the bbox.
+     * @param northeast The upper right coord of the bbox.
+     *
+     * @return the number of tiles removed
```

Comment:
```suggestion
     * Removes tiles wholly within the supplied bounding box. If the tile is not
     * contained completely within the bounding box, it will remain in the cache.
     * After removing files from the cache, any routers should be reconfigured
     * to synchronize their in-memory cache with the disk.
     *
     * @param tilePath The path to the tiles.
     * @param southwest The lower left coord of the bounding box.
     * @param northeast The upper right coord of the bounding box.
     *
     * @return the number of tiles removed
```